app/assets/javascripts/admin/addon/components/admin-report.gjs

```diff
@@ -0,0 +1,793 @@
+import didUpdate from "@ember/render-modifiers/modifiers/did-update";
+import { next } from "@ember/runloop";
+import { service } from "@ember/service";
+import { htmlSafe } from "@ember/template";
+import { isPresent } from "@ember/utils";
+import ConditionalLoadingSection from "discourse/components/conditional-loading-section";
+import DButton from "discourse/components/d-button";
+import DPageSubheader from "discourse/components/d-page-subheader";
+import DateTimeInputRange from "discourse/components/date-time-input-range";
+import concatClass from "discourse/helpers/concat-class";
+import dIcon from "discourse/helpers/d-icon";
+import number from "discourse/helpers/number";
+import { reportModeComponent } from "discourse/lib/admin-report-additional-modes";
+import { REPORT_MODES } from "discourse/lib/constants";
+import { bind } from "discourse/lib/decorators";
+import { isTesting } from "discourse/lib/environment";
+import { exportEntity } from "discourse/lib/export-csv";
+import { outputExportResult } from "discourse/lib/export-result";
+import { makeArray } from "discourse/lib/helpers";
+import ReportLoader from "discourse/lib/reports-loader";
+import { i18n } from "discourse-i18n";
+import AdminReportChart from "admin/components/admin-report-chart";
+import AdminReportCounters from "admin/components/admin-report-counters";
+import AdminReportInlineTable from "admin/components/admin-report-inline-table";
+import AdminReportRadar from "admin/components/admin-report-radar";
+import AdminReportStackedChart from "admin/components/admin-report-stacked-chart";
+import AdminReportStackedLineChart from "admin/components/admin-report-stacked-line-chart";
+import AdminReportStorageStats from "admin/components/admin-report-storage-stats";
+import AdminReportTable from "admin/components/admin-report-table";
+import ReportFilterBoolComponent from "admin/components/report-filters/bool";
+import ReportFilterCategoryComponent from "admin/components/report-filters/category";
+import ReportFilterGroupComponent from "admin/components/report-filters/group";
+import ReportFilterListComponent from "admin/components/report-filters/list";
+import Report, { DAILY_LIMIT_DAYS, SCHEMA_VERSION } from "admin/models/report";
+import DTooltip from "float-kit/components/d-tooltip";
+
+const TABLE_OPTIONS = {
+  perPage: 8,
+  total: true,
+  limit: 20,
+  formatNumbers: true,
+};
+
+const CHART_OPTIONS = {};
+
+export default class AdminReport extends Component {
+  @service siteSettings;
+
+  @tracked isEnabled = true;
+  @tracked isLoading = false;
+  @tracked rateLimitationString = null;
+  @tracked report = null;
+  @tracked model = null;
+  @tracked showTitle = true;
+  @tracked currentMode = null;
+  @tracked options = null;
+  @tracked dateRangeFrom = null;
+  @tracked dateRangeTo = null;
+
+  showHeader = this.args.showHeader ?? true;
+  showFilteringUI = this.args.showFilteringUI ?? false;
+  showDescriptionInTooltip = this.args.showDescriptionInTooltip ?? true;
```

Comment:
unless there is a specific reason to assign those at the initialization time, it's better to have those as getters, e.g.

```js
get showHeader() {
  return this.args.showHeader ?? true;
}
```

so that the component is reactive to the potential changes to its args

Reply:
None of these would change based on args, I think it's fine